### PR TITLE
add port PublishMode to service inspect --pretty output

### PR DESCRIPTION
**Pull request by dongluochen**
`PublishMode` (`ingress` or `host`) was added to port config in #27917. It should be added in service inspect pretty format.

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

File: cli/command/service/update.go [hunk 1]
```diff
@@ -694,7 +694,7 @@ portLoop:
 			if v, ok := portSet[portConfigToString(&port)]; ok {
 				if v != port {
 					fmt.Println("v", v)
-					return fmt.Errorf("conflicting port mapping between %v:%v/%s and %v:%v/%s", port.PublishedPort, port.TargetPort, port.Protocol, v.PublishedPort, v.TargetPort, v.Protocol)
+					return fmt.Errorf("conflicting port mapping between %v:%v/%s/%s and %v:%v/%s/%s", port.PublishedPort, port.TargetPort, port.Protocol, port.PublishMode, v.PublishedPort, v.TargetPort, v.Protocol, v.PublishMode)
```
Comment (aaronlehmann) [inline cli/command/service/update.go]: Wondering if this is the right notation. `127.0.0.1:80/tcp` is a standard notation; `127.0.0.1:80/tcp/ingress` looks a little weird. Maybe it should be `127.0.0.1:80/tcp (ingress)`?

Comment (dongluochen) [inline cli/command/service/update.go]: Actually this error message is misleading. `portConfigToString` now includes `PublishedPort, TargetPort, Protocol, PublishMode`. So this message would show `conflicting port mapping between A and A`. @vdemeester Shall we remove `if v != port` part here?

**Comment by aaronlehmann** [comment]
LGTM

It seems weird that this doesn't trigger an error when a user adds a port mapping with `docker service update` that duplicates an existing mapping. But this is the existing behavior, so it seems fine not to change it right now.

**Comment by dongluochen** [comment]
@vdemeester I want your feedback on if this is the right validation in port-add/rm.

File: cli/command/service/update.go (hunk 1 above)
Comment (vdemeester) [inline cli/command/service/update.go]: Ah you are right 👼

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by vdemeester** [comment]
Moving to docs reviews
@dongluochen I think `service_inspect.md` (in `docs/reference`) needs to be updated
/cc @thaJeztah

**Comment by thaJeztah** [comment]
yup, if you can update the output to match, that would be great; https://github.com/docker/docker/blob/master/docs/reference/commandline/service_inspect.md

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**